src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchElementViewModel.cs

```diff
@@ -41,6 +50,46 @@ public PackageManagerSearchElementViewModel(PackageManagerSearchElement element,
                 new DelegateCommand(() => GoToUrl(FormatUrl(Model.SiteUrl)), () => !String.IsNullOrEmpty(Model.SiteUrl));
             this.VisitRepositoryCommand =
                 new DelegateCommand(() => GoToUrl(FormatUrl(Model.RepositoryUrl)), () => !String.IsNullOrEmpty(Model.RepositoryUrl));
+
+        }
+
+        /// <summary>
+        /// Alternative constructor which takes a Function object in order to
+        /// assist communication between the PackageManagerSearchViewModel
+        /// and the PackageManagerSearchElementViewModel.
+        /// </summary>
+        /// <param name="element">A PackageManagerSearchElement</param>
+        /// <param name="canLogin">A Boolean used for access control to certain internal packages.</param>
+        /// <param name="isPackageInstalledFunction">A function that we pass to this constructor which takes a string and returns a Boolean.</param>
+        public PackageManagerSearchElementViewModel(PackageManagerSearchElement element, bool canLogin, Func<string,bool> isPackageInstalledFunction) : this(element, canLogin)
+        {
+            this.isPackageInstalled = isPackageInstalledFunction;
+            this.IsInstalled = isPackageInstalled(this.Model.Name);
+        }
+
+        private bool isInstalled;
+
+        /// <summary>
+        /// A Boolean flag reporting whether or not the user already has this SearchElement's package installed.
+        /// </summary>
+        public bool IsInstalled
+        {
+            get => isInstalled;
+            set
+            {
+                isInstalled = value;
+                RaisePropertyChanged(nameof(IsInstalled));
+            }
+        }
+
+        /// <summary>
+        /// Checks whether the user has already installed a package with the given name.
+        /// </summary>
+        /// <returns>A Boolean indicating whether the user already has a package installed with the given name.</returns>
+        internal bool IsPackageInstalled()
```

Comment:
I do not see the point of this method.
You could just use the public method IsInstalled instead.

Reply:
I'll agree this was a difficult one to get working, because there's not great visibility between the PackageManagerSearchElementViewModel and the PackageManagerSearchViewModel.

If there's a straightforward way for the PackageManagerSearchElement to determine whether it's installed, we could clear out a lot of code here.